File: src/main/java/com/connectflow/repository/PawnTransactionItemImageRepository.java
package com.connectflow.repository;

import com.connectflow.model.PawnTransactionItemImage;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;
import java.util.UUID;

@Repository
public interface PawnTransactionItemImageRepository extends JpaRepository<PawnTransactionItemImage, UUID> {
    List<PawnTransactionItemImage> findByItemIdOrderByImageOrderAsc(UUID itemId);
    List<PawnTransactionItemImage> findByTransactionId(UUID transactionId);
    void deleteByItemId(UUID itemId);
    void deleteByTransactionId(UUID transactionId);
}

File: next/backend/src/repository/PawnTransactionItemImageRepository.js
'use strict';
const { PawnTransactionItemImage } = require('../model');

module.exports = {
  findByItemId: (itemId) => PawnTransactionItemImage.findAll({ where: { itemId }, order: [['image_order', 'ASC']] }),
  findByTransactionId: (transactionId) => PawnTransactionItemImage.findAll({ where: { transactionId } }),
  create: (data) => PawnTransactionItemImage.create(data),
  deleteByItemId: (itemId) => PawnTransactionItemImage.destroy({ where: { itemId } }),
};
